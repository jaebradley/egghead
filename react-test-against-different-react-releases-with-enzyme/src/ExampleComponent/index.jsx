import React from 'react';
import Title from '../Title';

const ExampleComponent = () => (
  <div>
    <header className={"jae baebae"}>
      <img
        src={'https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg'}
        alt={'React logo'}
        />
      <h1>
        Blah blah blah I'm an example
      </h1>
    </header>
    <Title text={"Blah blah blah I'm a Title"} />
    <p className={"example-content"}>
      Blah blah blah I'm a p element
    </p>
  </div>
);

export default ExampleComponent;
