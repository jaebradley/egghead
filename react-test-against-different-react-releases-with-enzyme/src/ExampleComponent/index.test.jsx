import React from 'react';
import {
  shallow,
} from 'enzyme';

import ExampleComponent from './';

describe('<ExampleComponent />', () => {
  const wrapper = shallow(<ExampleComponent />);

  it('should contain 1 p element', () => {
    expect(wrapper.find('p').length).toBe(1);
  });

  it('should contain an element with class name "example content"', () => {
    expect(wrapper.find('.example-content').exists()).toBe(true);
  });

  it('should contain a header with two children', () => {
    expect(wrapper.find('header').children().length).toBe(2);
  });

  it('should contain a header with class name "jae"', () => {
    expect(wrapper.find('header').hasClass('jae')).toBe(true);
  });

  it('p element should contain correct text', () => {
    expect(wrapper.find('p').text()).toBe("Blah blah blah I'm a p element");
  });

  it('should contain title text', () => {
    expect(wrapper.find('[text="Blah blah blah I\'m a Title"]').text()).toBe("<Title />");
  });

  it('should contain title', () => {
    expect(wrapper.find('Title').exists()).toBe(true);
  });

  it('should find component with alt "React logo"', () => {
    expect(wrapper.find({ alt: 'React logo' }).exists()).toBe(true);
  });
});
