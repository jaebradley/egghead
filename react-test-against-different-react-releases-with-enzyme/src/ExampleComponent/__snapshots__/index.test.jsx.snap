// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExampleComponent /> matches the snapshot 1`] = `
<div>
  <header
    className="jae baebae"
  >
    <img
      alt="React logo"
      src="https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg"
    />
    <h1>
      Blah blah blah I'm an example
    </h1>
  </header>
  <Title
    text="Blah blah blah I'm a Title"
  />
  <p
    className="example-content"
  >
    Blah blah blah I'm a p element
  </p>
</div>
`;
